config/initializers/5_telemetry_metrics.rb

```diff
@@ -0,0 +1,12 @@
+# Foreman telemetry metrics registration.
+#
+# There are three types of telemetry measurements: counter, gauge, histogram. Each measurement has unique name
+# represented by a symbol, a description and list of tags which are required and mapped to monitoring frameworks
+# which do not support arbitrary tags.
+#
+telemetry = Foreman::Telemetry.instance
+telemetry.add_counter(:http_requests, 'A counter of HTTP requests made', [:controller, :action])
+telemetry.add_histogram(:http_request_total_duration, 'Total duration of controller action', [:controller, :action])
+telemetry.add_histogram(:http_request_db_duration, 'Time spent in database for a request', [:controller, :action])
+telemetry.add_histogram(:http_request_view_duration, 'Time spent in view for a request', [:controller, :action])
+telemetry.add_counter(:activerecord_instances, 'Number of instances of ActiveRecord models', [:class])
```

Comment:
Could the code in this file live under `lib/foreman/telemetry` ? If this is required to be initialized here, could the code live there but be a one line call out to it? I think it might be more obvious code wise to have everything under `lib/foreman/telemetry` witha simple call here like `telemetry.define_metrics`

Reply:
I moved Ruby and Rails out of initializers, but I'd like to keep main initialization and this file because this is the place core developers will register their own metrics. I am now adding also Foreman Plugin support so plugins can register their own - this will land in the next rebase.